src/shared/logger.ts

```diff
@@ -174,19 +173,31 @@ function writeToOutputChannel(messageLevel: number,
     }
 }
 
-// matches VS Code's log file name format
-// YYYYMMDDThhmmss (note the 'T' prior to time)
+// outputs a timestamp with the following formattings:
+// type: 'filename' = YYYYMMDDThhmmss (note the 'T' prior to time, matches VS Code's log file name format)
+// type: 'logFile' = YYYY-MM-DD HH:MM:SS
 // Uses local timezone
-function makeDateString(): string {
+function makeDateString(type: 'filename' | 'logfile'): string {
     const d = new Date()
+    const isFilename: boolean = type === 'filename'
 
-    return `${d.getFullYear()}` +
+    return `${d.getFullYear()}${chooseSpacer(isFilename, '', '-')}` +
     // String.prototype.padStart() is not available in Typescript...
-    `${(d.getMonth() + 1) < 10 ? '0' + (d.getMonth() + 1).toString() : (d.getMonth() + 1)}` +
-    `${d.getDay() < 10 ? '0' + d.getDay().toString() : d.getDay()}T` +
-    `${d.getHours() < 10 ? '0' + d.getHours().toString() : d.getHours()}` +
-    `${d.getMinutes() < 10 ? '0' + d.getMinutes().toString() : d.getMinutes()}` +
-    `${d.getSeconds() < 10 ? '0' + d.getSeconds().toString() : d.getSeconds()}`
+    `${padNumber(d.getMonth() + 1)}${chooseSpacer(isFilename, '', '-')}` +
+    `${padNumber(d.getDate())}${chooseSpacer(isFilename, 'T', ' ')}` +
+    `${padNumber(d.getHours())}${chooseSpacer(isFilename, '', ':')}` +
+    `${padNumber(d.getMinutes())}${chooseSpacer(isFilename, '', ':')}` +
+    `${padNumber(d.getSeconds())}`
+}
+
+function padNumber(num: number): string {
+    return num < 10 ? '0' + num.toString() : num.toString()
+}
+
+function chooseSpacer(isFilename: boolean, ifTrue: string, ifFalse: string): string {
```

Comment:
Any reason not to just use a ternary statement here? I.e.

```
isFilename ? '' : '-'
```

Instead of

```
chooseSpacer(isFilename, '', '-')
```

Reply:
Not really, I think I was just in a bit of a haze of refactoring and thought this would make more sense. Now that I see it written out, it's perfectly readable. I'll go ahead and add that.